.github/workflows/android-release.yml

```diff
@@ -0,0 +1,171 @@
+        uses:           actions/cache@v3
+        with:
+          path:         |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name:           Build
+        run:            |
+          ./gradlew :android:testDebugUnitTest
+
+  build:
+    name: Build signed APKs
+    runs-on: ubuntu-22.04
+    # if: github.ref == ' refs/head/main'
+    steps:
+      - name:           Cancel Previous Runs
+        uses:           styfle/cancel-workflow-action@b173b6ec0100793626c2d9e6b90435061f4fc3e5 # 0.11.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name:           Checkout Sources
+        uses:           actions/checkout@v3.5.0
+        with:
+          fetch-depth:  50
+          submodules:   'recursive'
+
+      - name:           Install System Dependencies (Linux)
+        if: runner.os == 'Linux'
+        run:            |
+          sudo apt-get update
+          sudo apt-get install -y clang libclang-dev libopencv-dev
+
+      - name:           Install Rust stable toolchain
+        uses:           actions-rs/toolchain@v1
+        with:
+          profile:      minimal
+          toolchain:    stable
+          override:     true
+
+      - name:                 Install uniffi_bindgen
+        uses:                 actions-rs/install@v0.1
+        with:
+          crate:              uniffi_bindgen
+          version:            0.22.0 # Must be in sync with version in Cargo.toml
+          use-tool-cache:     true
+
+      - name:           Install additional rust targets
+        run:            rustup target add aarch64-linux-android armv7-linux-androideabi x86_64-linux-android
+
+      - name:           Set up JDK environment
+        uses:           actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: 11
+
+      - name:           Install NDK
+        run:            echo "y" | sudo ${ANDROID_HOME}/tools/bin/sdkmanager --install "ndk;24.0.8215888" --sdk_root=${ANDROID_SDK_ROOT}
+
+      - name:           Rust Cache
+        uses:           Swatinem/rust-cache@6fd3edff6979b79f87531400ad694fb7f2c84b1f # v2.2.1
+        with:
+          workspaces:   rust -> rust/target
+
+      - name:           Gradle Cache
+        uses:           actions/cache@v3
+        with:
+          path:         |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      # Build and publish
+
+      - name:           Assemble release unsigned apk
+        run:            |
+          ./gradlew clean assembleRelease --stacktrace
+
+      - name: Sign app APK
+        uses: r0adkll/sign-android-release@v1
```

Comment:
This repository seems abandoned. Could we use another one?
https://github.com/r0adkll/sign-android-release

Reply:
It uses deprecated Node12, so one day it'll stop working this summer
https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/
https://github.com/r0adkll/sign-android-release/pull/70